File: Java/roman-to-integer.java
class Solution {
   
    public int romanToInt(String s) {

        HashMap<Character, Integer> romanMaps = new HashMap<>();
        romanMaps.put('A', 0);
        romanMaps.put('I', 1);
        romanMaps.put('V', 5);
        romanMaps.put('X', 10);
        romanMaps.put('L', 50);
        romanMaps.put('C', 100);
        romanMaps.put('D', 500);
        romanMaps.put('M', 1000);

        int result = 0;
        char[] chars = s.toCharArray();
        for (int i = 0; i < chars.length; i++) {
            char value = chars[i];
            if (i == chars.length - 1) {
                result += romanMaps.get(value);
                return result;
            } else {
                char value1 = chars[i + 1];
                if (romanMaps.get(value) >= romanMaps.get(value1)) {
                    result += romanMaps.get(value);
                } else {
                    result -= romanMaps.get(value);
                }
            }
        }
        return result;
    }
}

File: Kotlin/roman-to-integer.kt
class Solution {
    fun romanToInt(s: String): Int {
        val romanMap = mapOf('I' to 1, 'V' to 5, 'X' to 10, 'L' to 50, 'C' to 100, 'D' to 500, 'M' to 1000)
        var result : Int = 0
        for (i in 0 until s.lastIndex) {
            if (romanMap.getValue(s[i]) >= romanMap.getValue(s[i + 1])) {
                result += romanMap.getValue(s[i])
            } else {
                result -= romanMap.getValue(s[i])
            }
        }
        result += romanMap.getValue(s[s.lastIndex])
        return result  
    }
}